lib/email/receiver.rb

```diff
@@ -239,6 +244,18 @@ def wrap_body_in_quote(user_email)
       @body = "[quote=\"#{user_email}\"]\n#{@body}\n[/quote]"
     end
 
+    def create_post_action(email_log, type)
+      PostAction.act(email_log.user, email_log.post, type)
```

Comment:
Calling `act` doesn't seem to ensure the guardian calls are executed.

You'll need to make sure something like [this code](https://github.com/discourse/discourse/blob/master/app/controllers/post_actions_controller.rb#L10) is executed.

(you'll also need a test for this -- what happens when the guardian check fails?)